src/Robo/Plugin/Traits/SlackNotifierTrait.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Usher\Robo\Plugin\Traits;
+
+use GuzzleHttp\Client;
+use Robo\Result;
+
+/**
+ * Trait to provide notification functionality.
+ */
+trait SlackNotifierTrait
+{
+    /**
+     * The Tugboat dashboard URL.
+     *
+     * @var string
+     */
+    protected $tugboatDashboardUrl = 'https://dashboard.tugboatqa.com';
+
+    /**
+     * Notify Slack if a base preview build failed.
+     *
+     * @param \Robo\Result $result
+     *   The result of the task to check.
+     * @param bool $force
+     *   If the notification should be forced.
+     *
+     * @see https://docs.tugboatqa.com/starter-configs/code-snippets/slack-integration/
+     */
+    protected function notifySlackOnFailedBasePreviewBuild(Result $result, bool $force = false): void
+    {
+        // Confirm we are in a base preview.
+        if (!$this->isTugboatBasePreview() && !$force) {
+            return;
+        }
+        // If everything went well there is nothing to do.
+        if ($result->wasSuccessful() && !$force) {
+            $this->say('Skipping Slack notification since all tasks completed successfully.');
+            return;
+        }
+        // Build various variables and URLs for the Slack message.
+        $dashboard_url = sprintf(
+            '%s/%s',
+            $this->tugboatDashboardUrl,
+            getenv('TUGBOAT_PREVIEW_ID'),
+        );
+        $text = sprintf(
+            "Tugboat Base Preview failed to build for *%s*\n*Preview*: %s\n*Dashboard*: %s",
```

Comment:
@adamzimmermann  What do you think about making these proper markdown links so the notification isn't so verbose with URLs? Made it a list too, but I don't care as much about that.

```suggestion
            "Tugboat Base Preview failed to build for *%s:*\n* [Preview](%s)\n* [Dashboard](%s)",
```

Reply:
```suggestion
            "Tugboat [Base Preview](%s) failed to build for *%s*",
```
Should we go a step farther? The preview link isn't that valuable, the dashboard gets you where you need to go, and now it's all just one line.

Reply:
> The preview link isn't that valuable

Good point. What about a link to the PR though? Is that a link we can determine?

Reply:
Updated.